### PR TITLE
[FIX] mrp_partial_production: Missing quant reservation when producing t#11323

### PR DESCRIPTION
**Pull request by hugho-ad**
[FIX] mrp_partial_production: Missing quant reservation when producing
    less than the qty reserved
    
    Before this commit when you recorded production less than the qty you
    reserved on raw materials, the stock quants did not get reserved with
    the remaining qty from previous reservation but inventory lines.
    
    This causes a conflict when trying to cancel/unreserve the mrp
    production getting following error:
    
    It is not possible to unreserve more products of %s than you have in stock.
    
    After this commit, now quants reservation it's consistent with inventory
    reservation.

### CODE REVIEW

File: mrp_partial_production/models/stock.py [hunk 1]
```diff
@@ -28,10 +29,14 @@ def _split(self, qty, restrict_partner_id=False):
...
+            available_quantity += product_uom_qty
             aml_vals.append(vals)
         new_move_id = super(StockMove, self)._split(qty, restrict_partner_id)
         for vals in aml_vals:
             vals['move_id'] = new_move_id
             self.env['stock.move.line'].create(vals)
-        self.browse(new_move_id)._recompute_state()
+        new_move = self.browse(new_move_id)
+        new_move._update_reserved_quantity(
+            new_move.product_uom_qty, available_quantity, new_move.location_id, strict=False)
```
Comment (luisg123v) [inline mrp_partial_production/models/stock.py]: Are you sure it should be strict=False?

Comment (hugho-ad) [reply mrp_partial_production/models/stock.py]: yes, we got it from after debugging to following line
https://github.com/odoo/odoo/blob/11.0/addons/stock/models/stock_move.py#L950

**Review by luisg123v — APPROVED** [review]
LGTM, just provide a commit description, please.

**Comment by hugho-ad** [comment]
@luisg123v 

as we talk, now we ensure reserving inventory materials on strict mode since we should keep the same inventory when producing less than the reserved and avoid re-reserve on another child locations

**Comment by hugho-ad** [comment]
@keylor2906 @luisg123v  @JulioSerna 
please review

**Review by luisg123v — APPROVED** [review]
:+1:

CC @belisac

**Comment by keylor2906** [comment]
@hugho-ad Missing line at EOF, but LGTM

**Review by keylor2906 — APPROVED** [review]

**Comment by hugho-ad** [comment]
@keylor2906 Done!
thanks!

**Comment by hugho-ad** [comment]
@JulioSerna 

Your comments are welcome

**Comment by luisg123v** [comment]
This has been approved by the customer, so merging.

I'll squash because the 2nd commit is part of the reservation and it was a required change we noticed after the fix was implemented.

**Comment by hugho-ad** [comment]
:+1:

**Pull request merged**